From f381ec95320743ed36beadee18cca3aa28dc1a3f Mon Sep 17 00:00:00 2001
From: Hugo Duncan <hugo@hugoduncan.org>
Date: Fri, 21 Oct 2011 14:17:37 -0400
Subject: [PATCH] Add *enable-locals-clearing* to control locals clearing

When set to false, the canBeCleared flag in LocalBinding is set to false.
---
 src/jvm/clojure/lang/Compiler.java |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/jvm/clojure/lang/Compiler.java b/src/jvm/clojure/lang/Compiler.java
index 55e1268..01955db 100644
--- a/src/jvm/clojure/lang/Compiler.java
+++ b/src/jvm/clojure/lang/Compiler.java
@@ -238,6 +238,9 @@ static final public Var INSTANCE = Var.intern(Namespace.findOrCreate(Symbol.inte

 static final public Var ADD_ANNOTATIONS = Var.intern(Namespace.findOrCreate(Symbol.intern("clojure.core")),
                                             Symbol.intern("add-annotations"));
+static final public Var ENABLE_LOCALS_CLEARING = Var.intern(Namespace.findOrCreate(Symbol.intern("clojure.core")),
+                                                            Symbol.intern("*enable-locals-clearing*"), Boolean.TRUE).setDynamic();
+//boolean


 //Integer
@@ -5454,8 +5457,8 @@ public static class LocalBinding{
 	public final int idx;
 	public final String name;
 	public final boolean isArg;
-    public final PathNode clearPathRoot;
-	public boolean canBeCleared = true;
+	public final PathNode clearPathRoot;
+	public boolean canBeCleared = (Boolean)(ENABLE_LOCALS_CLEARING.deref());
 	public boolean recurMistmatch = false;

     public LocalBinding(int num, Symbol sym, Symbol tag, Expr init, boolean isArg,PathNode clearPathRoot)
--
1.7.5.4
